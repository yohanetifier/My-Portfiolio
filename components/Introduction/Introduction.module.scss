@import "../../styles/breakpoint.module.scss";

.wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100vw;
    height: 100vh;
    position: relative;
}

.imgWrapper,
.imgWrapper1,
.imgWrapper2,
.imgWrapper3,
.imgWrapper4 {
    opacity: 0;
}

.imgWrapper {
    border: 2px solid purple;
    position: absolute;
    width: 200px;
    height: 300px;
    transform: rotate(50deg);
    @include mobileS {
        width: 100px;
        height: 230px;
    }
}

.imgWrapper1 {
    border: 2px solid blue;
    position: absolute;
    width: 200px;
    height: 300px;
    transform: rotate(-20deg);
    @include mobileS {
        width: 100px;
        height: 230px;
    }
}

.imgWrapper2 {
    border: 2px solid orange;
    position: absolute;
    width: 400px;
    height: 500px;
    transform: rotate(30deg);
    @include tabletXs {
        width: 350px;
        height: 450px;
    }
    @include mobileM {
        width: 250px;
        height: 300px;
    }
    @include mobileS {
        width: 200px;
        height: 250px;
    }
}

.imgWrapper3 {
    border: 2px solid green;
    position: absolute;
    width: 350px;
    height: 350px;
    transform: rotate(10deg);
    @include mobileM {
        width: 300px;
        height: 300px;
    }
    @include mobileS {
        width: 200px;
        height: 200px;
    }
}

.imgWrapper4 {
    border: 2px solid white;
    position: absolute;
    width: 300px;
    height: 600px;
    @include tabletXs {
        width: 250px;
        height: 500px;
    }
    @include mobileM {
        width: 200px;
        height: 400px;
    }
    @include mobileS {
        width: 150px;
        height: 300px;
    }
}
