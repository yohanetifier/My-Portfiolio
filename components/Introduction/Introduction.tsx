import React, { RefObject, useState } from 'react';
import styles from './Introduction.module.scss';
import AnimatedWords from '../AnimatedWords/AnimatedWords';
import AnimatedImg from '../AnimatedImg/AnimatedImg';
import { ArrayOfImg } from '../../typings/ArrayofImg';

type SourceImage = Pick<ArrayOfImg, 'src' | 'alt'>;

type Props = {
	words: string[];
	images: SourceImage[];
	counter?: RefObject<number>;
	setHideIntro: React.Dispatch<React.SetStateAction<boolean>>;
};

const Introduction = ({ words, counter, setHideIntro }: Props) => {
	const [animationComplete, setAnimationComplete] = useState<boolean>(false);
	const imagesArray = [
		{
			id: 1,
			src: '/assets/images/poker.jpeg',
			alt: 'poker',
			className: styles.imgWrapper,
		},
		{
			id: 2,
			src: '/assets/images/plane.jpeg',
			alt: 'plane',
			className: styles.imgWrapper1,
		},
		{
			id: 3,
			src: '/assets/images/echecs.jpeg',
			alt: 'echecs',
			className: styles.imgWrapper2,
		},
		{
			id: 4,
			src: '/assets/images/fitness.jpeg',
			alt: '',
			className: styles.imgWrapper3,
		},
		{
			id: 5,
			src: '/assets/images/foot.jpeg',
			alt: '',
			className: styles.imgWrapper4,
		},
	];

	return (
		<div className={`${styles.wrapper} entranceWrapper`}>
			<AnimatedWords
				arrayOfLetter={words}
				setAnimationComplete={setAnimationComplete}
			/>
			<AnimatedImg
				arrayOfImg={imagesArray}
				animationComplete={animationComplete}
				counter={counter}
				setHideIntro={setHideIntro}
			/>
		</div>
	);
};

export default Introduction;
