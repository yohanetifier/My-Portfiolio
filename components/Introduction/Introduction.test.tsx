import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Introduction from './Introduction';

const words: string[] = ['H', 'E', 'L', 'L', 'O'];
const setHideIntro = jest.fn();
const images = [];

describe('Introduction', () => {
	it('should render correctly', () => {
		const { container } = render(
			<Introduction
				setHideIntro={setHideIntro}
				words={words}
				images={images}
			/>,
		);
		// expect(container).toMatchSnapshot();
	});
	it('should have a words', () => {
		const { getByText, getAllByText } = render(
			<Introduction
				setHideIntro={setHideIntro}
				words={words}
				images={images}
			/>,
		);
		const hLetter = getByText('H');
		const eLetter = getByText('E');
		const lLetters = getAllByText('L');
		const oLetter = getByText('O');
		expect(hLetter).toBeInTheDocument();
		expect(eLetter).toBeInTheDocument();
		expect(lLetters).toHaveLength(2);
		expect(oLetter).toBeInTheDocument();
	});
	it('should have 2 imgs', () => {
		const { getAllByRole } = render(
			<Introduction
				setHideIntro={setHideIntro}
				words={words}
				images={images}
			/>,
		);
		const imgs = getAllByRole('img');
		expect(imgs).toHaveLength(5);
	});
});
