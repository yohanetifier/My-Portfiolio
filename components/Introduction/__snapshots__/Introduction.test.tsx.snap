// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction should render correctly 1`] = `
<div>
  <div
    class="wrapper entranceWrapper"
  >
    <div
      class="title"
      data-testid="div"
    >
      <span
        class="letter"
        data-testid="span"
        style=""
      >
        H
      </span>
      <span
        class="letter"
        data-testid="span"
        style=""
      >
        E
      </span>
      <span
        class="letter"
        data-testid="span"
        style=""
      >
        L
      </span>
      <span
        class="letter"
        data-testid="span"
        style=""
      >
        L
      </span>
      <span
        class="letter"
        data-testid="span"
        style=""
      >
        O
      </span>
    </div>
    <div
      class="container"
    >
      <figure
        class="imgWrapper figure"
      >
        <img
          alt="poker"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fpoker.jpeg&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fpoker.jpeg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fpoker.jpeg&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </figure>
      <figure
        class="imgWrapper1 figure"
      >
        <img
          alt="plane"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fplane.jpeg&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fplane.jpeg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fplane.jpeg&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </figure>
      <figure
        class="imgWrapper2 figure"
      >
        <img
          alt="echecs"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Fechecs.jpeg&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Fechecs.jpeg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fechecs.jpeg&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </figure>
      <figure
        class="imgWrapper3 figure"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Ffitness.jpeg&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Ffitness.jpeg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Ffitness.jpeg&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </figure>
      <figure
        class="imgWrapper4 figure"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/_next/image?url=%2Fassets%2Fimages%2Ffoot.jpeg&w=640&q=75"
          srcset="/_next/image?url=%2Fassets%2Fimages%2Ffoot.jpeg&w=256&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Ffoot.jpeg&w=640&q=75 2x"
          style="color: transparent;"
          width="200"
        />
      </figure>
    </div>
  </div>
</div>
`;
